hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/NFTOwnersChangeStep.java

```diff
@@ -87,18 +87,12 @@ public void doIn(final TransferContext transferContext) {
                     validateSpenderHasAllowance(senderAccount, topLevelPayer, tokenId, nft);
                 }
 
-                final var copyNft = nft.copyBuilder();
-                // If the nft owner is not set then set it to the treasury account
-                if (!nft.hasOwnerId() || nft.ownerId().equals(AccountID.DEFAULT)) {
-                    copyNft.ownerId(treasury);
-                }
```

Comment:
Should we delete this ? Or just change the condition in `if` to see if owner is null ?